Lib/_pylong.py

```diff
@@ -277,44 +295,46 @@ def inner(n, w):
 
     # How many base 256 digits are needed?. Mathematically, exactly
     # floor(log256(int(s))) + 1. There is no cheap way to compute this.
-    # But we can get an upper buond, and that's necessary for our error
+    # But we can get an upper bound, and that's necessary for our error
     # analysis to make sense. int(s) < 10**len(s), so the log needed is
     # < log256(10**len(s)) = len(s) * log256(10). However, using
     # finite-precision floating point for this, it's possible that the
     # computed value is a little less than the true value. If the true
     # value is at - or a little higher than - an integer, we can get an
     # off-by-1 error too low. So we add 2 instead of 1 if chopping lost
     # a fraction > 0.9.
-    log_ub = len(s) * _LOG_10_BASE_256
+
+    # The "WASI" test platfrom can complain about `len(s)` if it's too
+    # large to fit in its idea of "an index-sized integer".
+    lenS = s.__len__()
+    log_ub = lenS * _LOG_10_BASE_256
     log_ub_as_int = int(log_ub)
-    w = log_ub_as_int + 1 + (log_ub - log_ub_as_int > 9.9)
-    # And what if we'vv plain exhausted the limits of HW floats? We
+    w = log_ub_as_int + 1 + (log_ub - log_ub_as_int > 0.9)
```

Comment:
Yes, the 9.9 constant confused me the most.

Reply:
Sorry about that! The typo gave a test that could never succeed. The difference is always in the semi-open internal `[0.0, 1.0)`